kratos/processes/generic_find_elements_neighbours_process.h

```diff
@@ -0,0 +1,341 @@
+    {
+        rOStream << "FindElementalNeighboursProcess";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ModelPart& mr_model_part;
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    void InitializeElementNeighbours()
+    {
+        block_for_each(mr_model_part.Elements(), [&](Element & rElement) {
+            const unsigned int nfaces = rElement.GetGeometry().FacesNumber();
+            if (rElement.Has(NEIGHBOUR_ELEMENTS)) {
+                ElementPointerVector& r_neighbour_elements = rElement.GetValue(NEIGHBOUR_ELEMENTS);
+                r_neighbour_elements.reserve(nfaces); 
+                r_neighbour_elements.erase(r_neighbour_elements.begin(),r_neighbour_elements.end() );
+            } else {
+                ElementPointerVector empty_vector;
+                empty_vector.reserve(nfaces); 
+                rElement.SetValue(NEIGHBOUR_ELEMENTS, empty_vector);
+            }
+        });
+
+    }
+
+
+    GlobalPointer<Element> CheckForNeighbourElems (const Geometry<Node<3> >& rFaceGeom,
+                                                 Element & rElement)
+    {
+        //creating a a vector of all the elem pointers
+        //therefore the elem of interest will be repeated as many times as nodes in the faces are
+        ElementPointerVector PointersOfAllFaceNodes;
+        const unsigned int nNodes = rFaceGeom.size();
+        for( unsigned int node_i = 0 ; node_i < nNodes; node_i++){ //starting from node 1
+            const ElementPointerVector& rNeighCandidates_i = rFaceGeom[node_i].GetValue(NEIGHBOUR_ELEMENTS);
+            for( unsigned int candidate_i_k = 0 ; candidate_i_k < rNeighCandidates_i.size(); candidate_i_k++){ //all against all
+                PointersOfAllFaceNodes.push_back(rNeighCandidates_i(candidate_i_k));
+            }
+        }
+
+        //the element itsef will also be present in the n nodes, so must be avoided
+        const unsigned int main_elem_id = rElement.Id();
+        //we will take the pointers in the first node as candidates
+        const unsigned int nCandidates =  rFaceGeom[0].GetValue(NEIGHBOUR_ELEMENTS).size();
+        const unsigned int nCompleteList = PointersOfAllFaceNodes.size();
+        for( unsigned int j = 0 ; j < nCandidates; j++){ 
+            unsigned int repetitions = 1 ; //the current one must be added
+            if(PointersOfAllFaceNodes[j].Id()!=main_elem_id){ //ignoring main node
+                for( unsigned int k = nCandidates ; k < nCompleteList; k++){ //starting from pointers belonging to the next node
+                    if(&(*PointersOfAllFaceNodes(j))==&(*PointersOfAllFaceNodes(k))){
```

Comment:
This comparison in particular can be problematic in MPI. If I understand correctly, you are comparing the physical address of the Element to remove duplicates. If you access by `&*` I assume because you want the address of the element to avoid having duplicated elements from different processes, you are calling the overloaded `*` from the global_pointer, which will segfault. use:

```C++
if( PointersOfAllFaceNodes(j).get() == PointersOfAllFaceNodes(k).get() ){
```

This will return the address without accessing the data.

Another solution will be to get the vector with the id's of the pointers via a communicator (like its done in the nodal_neighbor process)

Reply:
didnt know, thank you!